core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+    }
+
+    await this.clickOn(commitMessage);
+    await this.type(commitMessage, 'Testing Testing');
+    await this.clickOn('.e2e-test-save-draft-button');
+  }
+
+  /**
+   * This function checks whether changes has been drafted or not.
+   */
+  async expectExplorationToBeDraftedSuccessfully(): Promise<void> {
+    await this.page.waitForSelector('#tutorialSaveButton');
+    const button = await this.page.$('#tutorialSaveButton');
+
+    if (!button) {
+      throw new Error('Save button not found.');
+    }
+
+    const isDraftButtonDisabled = await button.evaluate(
+      button => (button as HTMLButtonElement).disabled
+    );
+
+    if (isDraftButtonDisabled) {
+      showMessage('Changes have been successfully drafted.');
+    } else {
+      throw new Error('Failed to draft changes.');
+    }
+  }
+
+  /**
+   * This function helps in publishing the exploration
+   */
+  async publishExploration(): Promise<void> {
+    await this.saveDraftExploration();
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(publishButtonDropDown);
+      await this.clickOn(mobilePublishButton);
+    } else {
+      await this.page.waitForSelector(`${publishButton}:not([disabled])`);
+      await this.clickOn(publishButton);
+    }
+    await this.page.waitForSelector(`${publishConfirmButton}:not([disabled])`);
+    await this.clickOn(publishConfirmButton);
+    await this.page.waitForSelector(closePublishedPopUp, {visible: true});
+    await this.clickOn(closePublishedPopUp);
+
+    explorationUrlAfterPublished = await this.page.url();
+  }
+
+  /**
+   * This function checks whether the exploration
+   * is published successfully or not.
+   */
+  async expectInteractionOnCreatorDashboard(): Promise<void> {
+    try {
+      await this.page.goto(explorationUrlAfterPublished);
+      showMessage('Exploration is available on creator dashboard.');
+    } catch (error) {
+      throw new Error('Failed to navigate to the exploration URL.');
+    }
+  }
+
+  /**
+   * This function is discarding the current changes.
+   */
+  async discardCurrentChanges(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(publishButtonDropDown);
+      await this.clickOn(mobileDiscardButton);
+    } else {
+      await this.clickOn(basicSettingsDropDown);
+      await this.clickOn('button.e2e-test-save-discard-toggle');
+      await this.page.waitForSelector(discardDraftButton, {visible: true});
+      await this.clickOn(discardDraftButton);
+    }
+    await this.page.waitForSelector('.e2e-test-confirm-discard-changes', {
+      visible: true,
+    });
+    await Promise.all([
+      this.clickOn(discardConfirmButton),
+      this.page.waitForNavigation(),
+    ]);
+    if (this.isViewportAtMobileWidth()) {
+      await this.clickOn(navBarOpener);
+      await this.clickOn(basicSettingsDropDown);
+    }
+  }
+
+  /**
+   *This function checks whether changes has discarded successfully or not.
```

Comment:
This doesn't make sense. The function should not know in which context it is being called. The docstring should just describe what the function does, in a simple way, without reference to what actually calls it in the top-level test.

Reply:
changed to `This function matches the expected title with added title`.